File: backend/src/main/java/de/hemfeinkost/backend/services/abstracts/PartyPlatterService.java
package de.hemfeinkost.backend.services.abstracts;

import de.hemfeinkost.backend.services.dtos.requests.PartyPlatterRequest;
import de.hemfeinkost.backend.services.dtos.responses.PartyPlatterCreatedResponse;
import de.hemfeinkost.backend.services.dtos.responses.PartyPlatterGetAllResponse;

import java.util.List;

public interface PartyPlatterService {
    List<PartyPlatterGetAllResponse> getAllPartyPlatters();
    List<PartyPlatterGetAllResponse> getActivePartyPlatters();
    PartyPlatterCreatedResponse getPartyPlatterById(long id);
    PartyPlatterCreatedResponse addPartyPlatter(PartyPlatterRequest partyPlatter);
    PartyPlatterCreatedResponse changePartyPlatterStatus(long id, boolean status);
}

File: frontend/src/services/PartyPlatterService.ts
import axios from "axios";
import {PartyPlatter} from "../types/PartyPlatter.ts";

export default class PartyPlatterService {
    getAllPartyPlatters() {
        return axios.get(`/api/party-platters`)
    }

    getActivePartyPlatters() {
        return axios.get(`/api/party-platters/active`)
    }

    getPartyPlatterById(id: number) {
        return axios.get(`/api/party-platters/${id}`)
    }

    addPartyPlatter(partyPlatter: PartyPlatter) {
        return axios.post(`/api/party-platters`, partyPlatter)
    }

    updatePartyPlatter(id: number, partyPlatter: PartyPlatter) {
        return axios.put(`/api/party-platters/${id}`, partyPlatter)
    }

    changePartyPlatterStatus(id: number, status: boolean) {
        return axios.put(`/api/party-platters/status/${id}?status=${status}`)
    }
}